src/west/cmd/project.py

```diff
@@ -124,22 +124,38 @@ def __init__(self):
         super().__init__(
             'branch',
             _wrap('''
-            Create topic branch.
+            Create a branch or list branches, in multiple projects.
 
             Creates a branch in each of the specified projects (default: all
-            cloned projects).
+            cloned projects). The new branches are set to track '{}'.
 
-            The new branches are set to track '{}'.
+            With no arguments, lists all local branches along with the
+            repositories they appear in.
 
             '''.format(_MANIFEST_REV_BRANCH) + _MANIFEST_REV_HELP))
 
     def do_add_parser(self, parser_adder):
-        return _add_parser(parser_adder, self, _branch_arg, _project_list_arg)
+        return _add_parser(parser_adder, self, _opt_branch_arg,
+                           _project_list_arg)
 
     def do_run(self, args, user_args):
-        for project in _projects(args):
-            if _cloned(project):
+        projects = [project for project in _projects(args) if _cloned(project)]
```

Comment:
I'm not asking for a change here, but I'm curious why you chose a list rather than a tuple comprehension here, since projects isn't mutated.

Reply:
Mostly handier syntax to be honest, though it's inconsistent. There are no tuple comprehension in Python.

The code below would turn `projects` into a generator instead, that can only be iterated over once (the parentheses are redundant, and the thing on the right is a *generator expression*):

    projects = (project for project in _projects(args) if _cloned(project))

Would have to make it something like this instead to get a tuple:

    projects = tuple(project for project in _projects(args) if _cloned(project))

In that case, the tuple constructor runs the generator to get the elements.

Reply:
> In that case, the tuple constructor runs the generator to get the elements.

Sorry for being sloppy; I did mean `tuple(generator expression)`.

Though since the variable is iterated over only once in the routine, the "bare" generator would work as well and not use any extra characters, no?

Reply:
Oh, right, forgot that it's only iterated once, just in separate paths.

Yeah, could use a bare generator then. Wonder if people would find that confusing though, since generator expressions are kinda obscure...

Reply:
Turns out you need the parentheses anyway.

Feels a bit risky in case someone thinks they can iterate it twice in the future, but could give it a try. ;)

Reply:
> Turns out you need the parentheses anyway.

Yeah, generator expressions are always parenethesized -- it's just that the same parentheses used for a function call etc. can do double duty for the generator.